lib/types/channels.d.ts

```diff
@@ -476,26 +470,37 @@ export interface URLButton extends ButtonBase {
     url: string;
 }
 
-export interface RawSelectMenu {
+export interface RawSelectMenuBase<T extends SelectMenuTypes> {
     custom_id: string;
     disabled?: boolean;
     max_values?: number;
     min_values?: number;
-    options: Array<SelectOption>;
     placeholder?: string;
-    type: ComponentTypes.SELECT_MENU;
+    type: T;
 }
 
-export interface SelectMenu {
+export type RawStringSelectMenu      = RawSelectMenuBase<ComponentTypes.STRING_SELECT> & { options: Array<SelectOption>; };
```

Comment:
interface

Reply:
wdym with this?

Reply:
Move the right side of the intersection into an interface